olx_importer/management/commands/load_components.py

```diff
@@ -146,35 +159,49 @@ def import_block_type(self, block_type, now, publish_log_entry):
         static_files_regex = re.compile(r"""['"]\/static\/(.+?)["'\?]""")
         block_data_path = self.course_data_path / block_type
 
-        for xml_file_path in block_data_path.glob('*.xml'):
+        for xml_file_path in block_data_path.glob("*.xml"):
             components_found += 1
             identifier = xml_file_path.stem
 
             # Find or create the Component itself
             component, _created = Component.objects.get_or_create(
                 learning_package=self.learning_package,
-                namespace='xblock.v1',
+                namespace="xblock.v1",
                 type=block_type,
                 identifier=identifier,
-                defaults = {
-                    'created': now,
-                }
+                defaults={
+                    "created": now,
+                },
             )
 
-            # Create the Content entry for the raw data...
+            # Create the RawContent entry for the raw data...
             data_bytes = xml_file_path.read_bytes()
             hash_digest = create_hash_digest(data_bytes)
-            data_str = codecs.decode(data_bytes, 'utf-8')
-            content, _created = Content.objects.get_or_create(
+
+            raw_content, created = RawContent.objects.get_or_create(
                 learning_package=self.learning_package,
-                mime_type=f'application/vnd.openedx.xblock.v1.{block_type}+xml',
+                mime_type=f"application/vnd.openedx.xblock.v1.{block_type}+xml",
                 hash_digest=hash_digest,
-                defaults = dict(
-                    data=data_bytes,
+                defaults=dict(
+                    # text=data_str,
                     size=len(data_bytes),
                     created=now,
-                )
+                ),
             )
+            if created:
+                raw_content.file.save(
+                    f"{raw_content.learning_package.uuid}/{hash_digest}",
+                    ContentFile(data_bytes),
+                )
```

Comment:
It's mostly consistency, and I thought it might make export simpler. I also thought that there might be some edge cases where we want to later promote RawContent to TextContent after the fact, and have them exist in both places–like video transcripts. But mostly I just like the idea that everything is minimally represented as an opaque blob, and there can be multiple levels of progressive enhancement of that data over time, e.g. XBlockContent, Video, etc.